app/scripts/modules/core/api/api.service.ts

```diff
@@ -23,6 +23,10 @@ export interface IRequestBuilder {
   put?: IRequestBuilder;
 }
 
+export interface IResponse {
+  ref: string;
+}
```

Comment:
I don't think this is right. The `ref` is specifically the response to the `triggerPipeline` method, and probably belongs in that service.